src/app/routes/registration/enrolment-form/enrolment-form.component.spec.ts

```diff
@@ -387,6 +388,40 @@ describe('EnrolmentFormComponent', () => {
       ?.controls[0];
     expect(formValue.value).toEqual('The label');
   });
+
+  it('should not update the form field if the copied input does not match the label', () => {
+    component.fieldList = [
+      {
+        fieldType: FieldTypesEnum.Text,
+        label: 'label 1',
+        required: true,
+      },
+    ];
+
+    component.toCopy = { 'label 2': 'The label' };
+
+    fixture.detectChanges();
+    const formValue = (component.enrolmentForm?.get('fields') as FormArray)
+      ?.controls[0];
+    expect(formValue.value).toBeFalsy();
```

Comment:
Is the `label 1` field falsy because it should never have been set and the default is a falsy value (whereas `'The label'` is truthy)?

Reply:
Algorithm is comparing labels. The problem is with setting value. We're passing at start empty object, getting keys (empty list), and first element. Later we're trying to find `undefined` in another list. Algorithm treat is as it exists, but it might not exist.
The test is checking what will happen if the label that i want to "copy" doesn't exist in fields.